### PR TITLE
Raster Attribute Table GUI enhancements/fixes

### LINKED ISSUE #51200: Unable to edit Raster Attribute Table values after adding column to new table
### What is the bug or the crash?

After making a new table for a raster layer, the fields for a newly created column cannot be updated.

### Steps to reproduce the issue

1. Download only the raster from https://noaa-ocs-nationalbathymetry-pds.s3.amazonaws.com/index.html#BlueTopo/BF2FW2KQ/ 
2. Add to QGIS
3. Update the Layer Styling to be Paletted/Unique values
4. Change the Band to Band3: Contributor and select Classify
5. Right click on the layer and select Create Raster Attribute Table and the Okay.
6. Make the table editable and add a column.
7. Try to add values to the newly created column.

### Versions

QGIS version
3.29.0-Master
QGIS code branch
master
Qt version
5.15.7
Python version
3.11.1
GDAL/OGR version
3.6.0
PROJ version
9.1.1
EPSG Registry database version
v10.076 (2022-08-31)
GEOS version
3.11.1-CAPI-1.17.1
SQLite version
3.36.0
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.5
QScintilla2 version
2.13.0
OS version
Windows 10 Version 2009

This copy of QGIS writes debugging output.

Active Python plugins

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by elpaso**
Fixes #51200 and more small GUI/UX related glitches:

- locale aware double input (consistent behavior with the vector attr table and the widgets)
- locale aware double formatting (same logic of the vector fields of type double)
- right-alignment of double fields
- wrong tooltip for last column

### CODE REVIEW

**Comment by elpaso** [comment]
CC @manisandro mingw64 builds seem broken (protobuf missing), would you please have a look?

**Comment by manisandro** [comment]
@elpaso Should resolve itself within a day or two, the new protobuf is already built.

**Comment by elpaso** [comment]
> @elpaso Should resolve itself within a day or two, the new protobuf is already built.

Thank you!

File: src/gui/raster/qgsrasterattributetablemodel.cpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ QString QgsRasterAttributeTableModel::headerTooltip( const int section ) const
   }
 
   const QString fieldName { hNames.at( section ) };
-  const bool isColor { section == hNames.count( ) - 1 };
+  const bool isColor { hasColor() &&section == hNames.count( ) - 1 };
```
Comment (nyalldawson) [inline src/gui/raster/qgsrasterattributetablemodel.cpp:75]:
```suggestion
  const bool isColor { hasColor() && section == hNames.count( ) - 1 };
```

File: src/gui/raster/qgsrasterattributetablemodel.cpp [hunk 2]
```diff
@@ -330,14 +330,14 @@ QVariant QgsRasterAttributeTableModel::data( const QModelIndex &index, int role
   if ( mRat && index.isValid() && index.row() < rowCount( QModelIndex() ) && index.column() < columnCount( QModelIndex() ) )
   {
     const QString fieldName { headerNames().at( index.column() ) };
-    const bool isColorOrRamp { index.column() == columnCount( QModelIndex() ) - 1 };
+    const bool isColorOrRamp { ( hasColor() || hasRamp() ) &&index.column() == columnCount( QModelIndex() ) - 1 };
```
Comment (nyalldawson) [inline src/gui/raster/qgsrasterattributetablemodel.cpp:333]:
```suggestion
    const bool isColorOrRamp { ( hasColor() || hasRamp() ) && index.column() == columnCount( QModelIndex() ) - 1 };
```

**Review by nyalldawson — APPROVED** [review]

**Comment by elpaso** [comment]
@nyalldawson thanks for the review, I think astyle is getting in the way. I am pretty sure I actually put spaces as you did but they were removed by astyle during pre-commit hooks (and I didn't notice).

**Comment by nyalldawson** [comment]
@elpaso maybe it's time we moved to something more modern!

**Comment by elpaso** [comment]
> @elpaso maybe it's time we moved to something more modern!

clang-format is the answer!

**Pull request merged**